### PR TITLE
Docstrings improved in calculus/singularities.py

### LINKED ISSUE #15621: README should refer to docs/README.rst for Dependencies
The [README](https://github.com/AnimeshSinha1309/SympyRepo/blob/master/README.rst) file in the repository suggests that we generate the docs using `make html` right away, but the docs do not compile. The file **docs/README.rst** includes what packages we need to install to compile the documentation.

This should be mentioned in the main _README_, that we go to _docs/README_ and check for the packages. Will make it easier to start off quick.

### PR DESCRIPTION
**Pull request by AnimeshSinha1309**
#### References to other Issues or PRs
Partially improves #15370 - Only for the file calculus/singularities.py

#### Brief description of what is fixed or changed
* Changed all the doctrings in the module calculus.singularities to the numpydoc format.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by moorepants** [comment]
This looks great. +1 from me if the tests pass.

File: doc/README.rst [hunk 1]
```diff
@@ -16,6 +16,10 @@ and to view it, do::
 
     firefox _build/html/index.html
 
+If you get **mpmath** error, install python3-mpmath package::
+
+    apt-get install python3-mpmath
+
```
Comment (moorepants) [inline doc/README.rst]: Please move these to a different PR.

File: README.rst [hunk 2]
```diff
@@ -63,6 +63,9 @@ You can generate everything at the above site in your local copy of SymPy by::
     $ cd doc
     $ make html
 
+To fulfill the package requirement, ensure that all the dependencies for building
+documentation enlisted in `doc/README.rst` are satisfied.
+
```
Comment (moorepants) [inline README.rst]: move to different PR

File: doc/README.rst (hunk 1 above)
Comment (AnimeshSinha1309) [reply doc/README.rst]: Sure will do, don't know how to move them out but I guess I can make another commit undoing these changes, and then redo them from another branch.

File: README.rst (hunk 2 above)
Comment (AnimeshSinha1309) [reply README.rst]: Okay, did move both these files to #15626, Deleted those commits here. Sorry for the force push, but knew no other way.

File: sympy/calculus/singularities.py [hunk 3]
```diff
@@ -44,6 +53,21 @@ def singularities(expression, symbol):
     >>> singularities(1/(y**3 + 1), y)
     {-1, 1/2 - sqrt(3)*I/2, 1/2 + sqrt(3)*I/2}
 
+    Returns
+    =======
+
+    Set
+        A set of values for `symbol` for which `expression` has a
+        singularity. An `EmptySet` is returned if `expression` has no
+        singularities for any given value of `Symbol`.
```
Comment (moorepants) [inline sympy/calculus/singularities.py]: Note that in RestructuredText if you want monospaced font you need to do: ` ``my text`` `

**Comment by AnimeshSinha1309** [comment]
@moorepants can you please check and tell me if I am done with everything. It this ready to be merged?

File: sympy/calculus/singularities.py [hunk 4]
```diff
@@ -28,6 +28,15 @@ def singularities(expression, symbol):
     Currently supported functions are:
     - univariate rational (real or complex) functions
```
Comment (moorepants) [inline sympy/calculus/singularities.py]: Things like this can be in the "Notes" section below.

File: sympy/calculus/singularities.py [hunk 5]
```diff
@@ -189,6 +297,13 @@ def is_strictly_decreasing(expression, interval=S.Reals, symbol=None):
     >>> is_strictly_decreasing(-x**2 + y, Interval(-oo, 0), x)
     False
 
+    Returns
+    =======
+
+    Boolean
+        True if ``expression`` is strictly decreasing in the given ``interval``,
+        False otherwise.
```
Comment (moorepants) [inline sympy/calculus/singularities.py]: Examples should come after "Returns". The first two sections should be "Parameters" and "Returns". I think the numpydoc spec gives the best ordering of sections.

File: sympy/calculus/singularities.py [hunk 6]
```diff
@@ -197,6 +312,13 @@ def is_monotonic(expression, interval=S.Reals, symbol=None):
     """
     Return whether the function is monotonic in the given interval.
 
+    expression: Expr
+        The target function which is being checked.
+    interval: Symbol
+        The range of values in which we are testing, defaults to all reals.
+    symbol: Symbol
```
Comment (moorepants) [inline sympy/calculus/singularities.py]: One thing to check is whether this strictly has to be a Symbol. Often a Function might work or even sometimes an expression. I don't know how restrictive these functions are but many sympy functions aren't as restrictive. So try to check the restrictions of input type and then you can do "Symbol or Function" or "sympyfiable" if sympify is called internal to the function.

**Comment by AnimeshSinha1309** [comment]
I have fixed the order of items, and moved the thing to Notes. I think I got the return types right.
All expressions we are checking have to be `Expr`, does not make sense otherwise. And then interval has to be a `Set`, that is the most flexible abstract class *(changed this one)*. And finally symbols, I don't see how they can be anything but `Symbol`, @moorepants please tell me if I am wrong on this.

**Comment by moorepants** [comment]
Looks great. Thanks for the fix!
